### PR TITLE
Fixing local ensemble download to set maximum parallel downloads

### PR DESCRIPTION
**Pull request by mmerce**

### CODE REVIEW

File: lib/LocalEnsemble.js [hunk 1]
```diff
@@ -99,17 +101,20 @@ function LocalEnsemble(ensembleOrModels, connection, maxModels) {
    */
   var i, fillEnsembleInfo, fillModelsInfo, fillSeparateModelsInfo, fill,
     addModel, self = this, ensemble, resource, filename, storageError,
-    localModel;
+    localModel, fillModelsBatch;
   if (!maxModels) {
     maxModels = MAX_MODELS;
   }
+  if (!maxParallelDownloads) {
```
Comment (sdesimone) [inline lib/LocalEnsemble.js:108]: shouldn't this be something like: `if (!maxParallelDownloads || maxParallelDownloads > MAX_PARALLEL_DOWNLOADS)`? If so, this remark would apply to line #105 as well

File: lib/LocalEnsemble.js [hunk 2]
```diff
@@ -119,71 +124,86 @@ function LocalEnsemble(ensembleOrModels, connection, maxModels) {
...
-    }
-    if (self.boosting) {
-      if (self.isRegression) {
-        self.boostingOffsets = (resource['initial_offset'] ||
-          REGRESSION_BOOSTING_OFFSET_DFT);
-      } else {
-        self.boostingOffsets = toMap(resource['initial_offsets'] ||
-          CLASSIFICATION_BOOSTING_OFFSET_DFT);
+      if (self.modelsSplits.length == 0) {
+        batches = Math.ceil((self.modelIds.length + 0.0) / self.maxModels);
```
Comment (sdesimone) [inline lib/LocalEnsemble.js:152]: Is this `+ 0.0` here due to some JS quirk? what is it for?

File: lib/LocalEnsemble.js (hunk 1 above)
Comment (mmerce) [reply lib/LocalEnsemble.js:108]: It depends. As I added it, the maxParallelDownloads can be set higher than MAX_PARALLEL_DOWNLOADS and it's the programmer's responsibility if it does not work then. As you suggest, MAX_PARALLEL_DOWNLOADS is not a default but a hard limit. We can do that too, but then we might want to find out which is the highest limit that we can afford, right? Maybe 10 is a bit low...

File: lib/LocalEnsemble.js (hunk 2 above)
Comment (mmerce) [reply lib/LocalEnsemble.js:152]: No problem, both seem correct.

**Review by sdesimone — APPROVED** [review]

**Pull request merged**